iroh/src/downloader.rs

```diff
@@ -237,6 +241,12 @@ impl Downloader {
 
             let service = Service::new(getter, dialer, concurrency_limits, msg_rx);
 
+            #[cfg(feature = "log-self")]
+            {
+                service.run().instrument(trace_span!("downloader", %me))
```

Comment:
same here, it should always attach the span and be `info_span` as it's run in a task.  Again try to do it around the entire spawned future and not just the `service.run()` method.